far/editor.cpp

```diff
@@ -4047,16 +4047,26 @@ void Editor::SaveFoundItemsToNewEditor(const VMenu& ListBox, const bool Matching
 
 string Editor::GetSearchAllFileName() const
 {
-	auto SearchAllFileName{ msg(lng::MEditSearchAllFileNameAppend) };
+	string MyFilename;
+	string_view Stem;
+	string_view Ext;
+	auto MsgId{ lng::MEditSearchAllFileNameFormat };
 
-	const auto HostFileEditor = GetHostFileEditor();
-	if (!HostFileEditor) return SearchAllFileName;
+	if (const auto MyFileEditor = GetHostFileEditor())
+	{
+		string MyType;
+		MyFileEditor->GetTypeAndName(MyType, MyFilename);
+		const auto [MyStem, MyExt]{ name_ext(MyFilename) };
+
+		Stem = MyStem;
+		if (MyExt.size() > 1) // More than empty or "."
+			Ext = MyExt;
+	}
 
-	string HostType, HostName;
-	HostFileEditor->GetTypeAndName(HostType, HostName);
+	if (!Ext.empty() && contains(EdOpt.SearchAllUseAltFileNameFormat.Get(), Ext))
```

Comment:
1. EdOpt.SearchAllUseAltFileNameFormat = `".txt;.csv"`
2. Ext = `".cs"`
3. ?

Reply:
Right! Let me fix it. Thanks.

Reply:
You can use `enum_tokens` here.

Reply:
Thank you for the hint. I am experimenting with ranges. If it works and not obscenely expensive, I'll go with it, if you do not mind. Likely later today my time (I have to step away right now).